hidi/factorization.py

```diff
@@ -0,0 +1,202 @@
+import warnings
+import numpy as np
+import collections
+from numpy.random import permutation
+from sklearn.decomposition import TruncatedSVD
+from hidi.transform import Transform
+
+# Catch annoying warnings from nimfa
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    import nimfa as nf
+
+
+class W2VStringTransform(Transform):
+    """
+    Takes a pandas Dataframe and transforms it into a
+    string
+
+    :param n_shuffles: The number of suffles for the
+    `item_id`.
+    :type n_shuffles: int
+    """
+    def __init__(self, n_shuffles=3, **w2v_kwargs):
+        self.w2v_kwargs = w2v_kwargs
+        self.n_shuffles = n_shuffles
+
+    def transform(self, df_temp, **kwargs):
+        """
+        :param df: a pandas Dataframe with two columns:
+        :code:`link_id`, :code:`item_id`
+        :type df: pandas.Dataframe
+        :rtype: a string of shuffled item_id
+        """
+        df = df_temp
```

Comment:
This looks suspicious. Is the intent here to copy `df_temp`? If so, I think this only copies the reference to the underlying dataframe object. To actually copy it, one needs to do:

```python
df  = df_temp.copy(deep=deep)
```

Where deep is a boolean.